src/components/Leaderboard/LeaderboardEntry.style.ts

```diff
@@ -0,0 +1,187 @@
+      }),
+    ...(isUserPosition && {
+      '&:before': {
+        content: '""',
+        position: 'absolute',
+        top: -1,
+        bottom: -1,
+        left: -2,
+        right: -2,
+        borderRadius: '6px',
+        backgroundColor: alpha(theme.palette.black.main, 0.04),
+        boxShadow: `inset 0 0 0 1px ${theme.palette.grey[400]}`,
+        [theme.breakpoints.up('sm' as Breakpoint)]: {
+          left: -12,
+          right: -12,
+        },
+      },
+    }),
+  }),
+);
+
+export const LeaderboardEntryDivider = styled(Divider)(({ theme }) => ({
+  color: theme.palette.alphaDark100.main,
+  margin: theme.spacing(0, 3),
+}));
+
+export const LeaderboardEntryInfos = styled(Box)(({ theme }) => ({
+  display: 'flex',
+  alignItems: 'center',
+  maxWidth: 'calc( 100% - 80px)',
+  gap: theme.spacing(1),
+  [theme.breakpoints.up('sm' as Breakpoint)]: {
+    gap: theme.spacing(4),
+    maxWidth: '80%',
+  },
+}));
+
+export const LeaderboardEntryConnect = styled(ButtonSecondary)(({ theme }) => ({
+  color: '#200052',
+  height: 40,
+}));
+
+export const RankLabel = styled(Typography)(({ theme }) => ({
+  padding: theme.spacing(1),
+  display: 'inline-block',
+  backgroundColor: alpha(theme.palette.text.primary, 0.04),
+  borderRadius: '16px',
+  minWidth: 32,
+  textAlign: 'center',
+  [theme.breakpoints.up('sm' as Breakpoint)]: {
+    padding: theme.spacing(1),
+  },
+}));
+
+export const RankLabelSkeleton = styled(Skeleton)(({ theme }) => ({
+  padding: theme.spacing(1),
+  display: 'inline-block',
+  borderRadius: '16px',
+  height: 24,
+  width: 48,
+  minWidth: 84,
+  [theme.breakpoints.up('sm' as Breakpoint)]: {
+    height: 32,
+    padding: theme.spacing(1, 2),
+    minWidth: 100,
+  },
+}));
+
+interface RankWalletProps extends ImageProps {
+  isUserEntry?: boolean;
+}
+
+export const RankWalletImage = styled(Image, {
+  shouldForwardProp: (prop) => prop !== 'isUserEntry',
+})<RankWalletProps>(({ theme, isUserEntry }) => ({
+  ...(isUserEntry && {
+    display: 'none',
+  }),
+  borderRadius: '100%',
+  width: 24,
+  height: 24,
+  [theme.breakpoints.up('sm' as Breakpoint)]: {
+    display: 'block',
+    width: 48,
+    height: 48,
+  },
+}));
+
+export const RankWalletImageSkeleton = styled(Skeleton)(({ theme }) => ({
+  borderRadius: '100%',
+  display: 'none',
+  width: 24,
+  height: 24,
+  flexShrink: 0,
+  [theme.breakpoints.up('sm' as Breakpoint)]: {
+    display: 'block',
+    width: 48,
+    height: 48,
+  },
+}));
+
+interface RankWalletAddressProps extends TypographyProps {
+  isUserConnected?: boolean;
+  isUserEntry?: boolean;
+}
+
+export const RankWalletAddress = styled(Typography, {
+  shouldForwardProp: (prop) =>
+    prop !== 'isUserConnected' && prop !== 'isUserEntry',
+})<RankWalletAddressProps>(({ theme, isUserConnected, isUserEntry }) => ({
+  textOverflow: 'ellipsis',
+  overflow: 'hidden',
+  [theme.breakpoints.down('sm' as Breakpoint)]: {
+    fontSize: '14px',
+    ...(!isUserConnected &&
```

Comment:
Would be better to have 1 variable within statement into it (2 variables are forwarded for only 1 reason)

Reply:
This may not be clear at first sight, but all three entries serve their purpose. Generally, that component "LeaderboardEntry" is re-used not only within the Leaderboard itself, but also above the leaderboard.
![image](https://github.com/user-attachments/assets/d04c1d83-5ec3-46ec-b9ee-be73246f7979)

- **isUserEntry** --> Handles the upper component above the Leaderboard
- **isUserConnected** --> Needed to distinguish different states / layouts on the above ( either show entry details or return the connect button)
- **isUserPosition**  --> Handles the actual user´s position within the Leaderboard

Reply:
But there's only 1 change for 2 specific variables, just do something like 

<Component isSomething={!isUserConnected && isUserEntry} />

And that will solve it without the needs of creating some more variables (that will not be used somewhere else than for that purpose)

Reply:
Ah, well I see you were referring to the component  <RankWalletAddress> and not the whole leaderboard overall!
It´s refactored now.